x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/TransformProgressGatherer.java

```diff
@@ -49,10 +51,20 @@ public static void getInitialProgress(Client client,
     public static SearchRequest getSearchRequest(DataFrameTransformConfig config) {
         SearchRequest request = new SearchRequest(config.getSource().getIndex());
         request.allowPartialSearchResults(false);
+        BoolQueryBuilder existsClauses = QueryBuilders.boolQuery();
+        config.getPivotConfig()
```

Comment:
[optional]
IMO in this case the old good foreach syntax is both shorter and more readable:
for (SingleGroupSource src : config.getPivotConfig().getGroupConfig().getGroups().values()) {
  existsClauses.must(QueryBuilders.existsQuery(src.getField())));
}

But leaving the choice up to you.